consensus/votepower/roster.go

```diff
@@ -61,81 +62,95 @@ func (b Ballot) String() string {
 
 // NewRound ..
 func NewRound() *Round {
-	return &Round{AggregatedVote: nil, BallotBox: map[string]*Ballot{}}
+	return &Round{
+		AggregatedVote: &bls.Sign{},
+		BallotBox:      map[shard.BlsPublicKey]*Ballot{},
+	}
 }
 
-type stakedVoter struct {
-	IsActive         bool               `json:"is-active"`
-	IsHarmonyNode    bool               `json:"is-harmony"`
-	EarningAccount   common.Address     `json:"earning-account"`
-	Identity         shard.BlsPublicKey `json:"bls-public-key"`
-	RawPercent       numeric.Dec        `json:"voting-power-unnormalized"`
-	EffectivePercent numeric.Dec        `json:"voting"`
-	EffectiveStake   numeric.Dec        `json:"effective-stake"`
+// PureStakedVote ..
+type PureStakedVote struct {
+	EarningAccount common.Address     `json:"earning-account"`
+	Identity       shard.BlsPublicKey `json:"bls-public-key"`
+	GroupPercent   numeric.Dec        `json:"group-percent"`
+	EffectiveStake *big.Int           `json:"effective-stake"`
 }
 
-// Roster ..
-type Roster struct {
-	Voters                          map[shard.BlsPublicKey]stakedVoter
+// AccommodateHarmonyVote ..
+type AccommodateHarmonyVote struct {
+	PureStakedVote
+	IsHarmonyNode  bool        `json:"-"`
+	OverallPercent numeric.Dec `json:"overall-percent"`
+}
+
+// String ..
+func (v AccommodateHarmonyVote) String() string {
+	s, _ := json.Marshal(v)
+	return string(s)
+}
+
+type topLevelRegistry struct {
 	OurVotingPowerTotalPercentage   numeric.Dec
 	TheirVotingPowerTotalPercentage numeric.Dec
-	RawStakedTotal                  numeric.Dec
-	HmySlotCount                    int64
+	RawStakedTotal                  *big.Int
```

Comment:
So because RawStakedTotal is not related to protocol. You changed it from dec to int for simplicity?

Reply:
Because I realized that stake can only be integer because token is only integer and leaving as big int reduces memory usage and your point as well